frontend-dev/src/components/AllIntegrations/GoogleSheet/GoogleSheetAuthorization.jsx

```diff
@@ -81,19 +84,22 @@ export default function GoogleSheetAuthorization({
 
 
   useEffect(() => {
-    if (step === 1) {
-      if (isEdit) {
-        const authIdExists = authData.find(auth => auth.id === sheetConf.authId);
-
-        if (sheetConf.authId && authIdExists) {
-          setSelectedUserId(sheetConf.authId)
-        } else if (authIdExists === false) {
-          setSelectedUserId(null)
-        }
+
+    if (step === 1 && isEdit) {
+
+      const updatedConf = { ...sheetConf }
```

Comment:
unnecessary spread operation,

Spreading an object creates a shallow copy of that object, which allows you to modify the copy without affecting the original object. This is important when you want to follow immutability principles. In immutability, instead of changing an object's properties directly, you create a new object with the modified values.